version: '3.7'

services:
  python_main:
    image: python_container
    container_name: finance_python
    build:
      context: .
      dockerfile: lib/docker/Dockerfile
    environment:
      AIRBYTE_USER: ${AIRBYTE_USER}
      AIRBYTE_PASSWORD: ${AIRBYTE_PASSWORD}
      AIRBYTE_HOST: ${AIRBYTE_HOST}
      SUPERSET_HOST: ${SUPERSET_HOST}
      SUPERSET_USER: ${SUPERSET_USER}
      SUPERSET_PASSWORD: ${SUPERSET_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_HOST: ${POSTGRES_HOST}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      PYTHONPATH: ${PYTHONPATH}
    networks:
      - backend
      - my_custom_network
    depends_on:
      - postgres_finance
    volumes:
      - ./lib:/finance_app/lib

  postgres_finance:
    image: postgres:15
    build:
      context: .
      dockerfile: lib/postgres/Dockerfile
    container_name: personal_finance_database
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - "6543:5432"
    volumes:
      - postgres-data:/var/lib/postgresql/data
      - ./lib/postgres/sql/init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - backend


volumes:
  postgres-data:

networks:
  backend:
    driver: bridge
  my_custom_network:
    external: true